### PR TITLE
feat: Enable Organizer to resend invitation

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #https://github.com/fossasia/open-event-frontend/issues/5416

require for https://github.com/fossasia/open-event-frontend/pull/6091

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/custom/role_invites.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+role_invites_routes_blueprint = Blueprint('role_invites_routes', __name__, url_prefix='/v1/role-invites')
+
+@role_invites_routes_blueprint.route(
+    '/<int:role_invite_id>/resend-invite'
+)
+@jwt_required
+def resend_invite(role_invite_id):
+    role_invite = RoleInvite.query.get_or_404(role_invite_id)
+    user = User.query.filter_by(email=role_invite.email).first()
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:22]: Duplicate logic
As discussed, move to a function in role invite model

File: app/api/custom/role_invites.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+from app.api.helpers.errors import ConflictError, ForbiddenError, NotFoundError
+from app.api.helpers.mail import send_email_role_invite, send_user_email_role_invite
+from app.api.helpers.permissions import jwt_required
+from app.api.helpers.notification import send_notif_event_role
+from app.api.helpers.permission_manager import has_access
+from app.models.event import Event
+from app.models.role_invite import RoleInvite
+from app.models.user import User
+
+role_invites_routes_blueprint = Blueprint('role_invites_routes', __name__, url_prefix='/v1/role-invites')
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:14]:
```suggestion
role_invites_routes = Blueprint('role_invites_routes', __name__, url_prefix='/v1/role-invites')
```

**Comment by sachinchauhan2889** [comment]
@iamareebjamal ok sir, i will do this changes tomorrow.

File: app/api/custom/role_invites.py [hunk 3]
```diff
@@ -3,40 +3,24 @@
...
 @jwt_required
 def resend_invite(role_invite_id):
     role_invite = RoleInvite.query.get_or_404(role_invite_id)
-    user = User.query.filter_by(email=role_invite.email).first()
     event = Event.query.filter_by(id=role_invite.event_id).first()
-    frontend_url = get_settings()['frontend_url']
-    link = "{}/e/{}/role-invites?token={}".format(
-        frontend_url, event.identifier, role_invite.hash
-    )
     if has_access('is_organizer', event_id=event.id):
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:18]: This should be inside the method as well

File: app/api/custom/role_invites.py [hunk 4]
```diff
@@ -3,40 +3,24 @@
...
-        if user:
-            send_user_email_role_invite(
-                role_invite.email, role_invite.role_name, event.name, link
-            )
-            send_notif_event_role(user, role_invite.role_name, event.name, link, event.id)
-        else:
-            send_email_role_invite(
-                role_invite.email, role_invite.role_name, event.name, link
-            )
+        RoleInvite.send_invite_mail(role_invite)
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:19]: Why are you calling it like this? How do you call a function on an object in JS? Python is also object oriented

File: app/models/role_invite.py [hunk 5]
```diff
@@ -47,3 +52,21 @@ def __repr__(self):
         return '<RoleInvite {!r}:{!r}:{!r}>'.format(
             self.email, self.event_id, self.role_id
         )
+
+
+    def send_invite_mail(self):
```
Comment (iamareebjamal) [inline app/models/role_invite.py:57]: Just send_invite

File: app/api/custom/role_invites.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+role_invites_routes = Blueprint('role_invites_routes', __name__, url_prefix='/v1/role-invites')
+
+@role_invites_routes.route(
+    '/<int:role_invite_id>/resend-invite'
+, methods=['POST'])
+@jwt_required
+def resend_invite(role_invite_id):
+    role_invite = RoleInvite.query.get_or_404(role_invite_id)
+    event = Event.query.filter_by(id=role_invite.event_id).first()
+    if has_access('is_organizer', event_id=event.id):
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:18]: Use is_coorganizer

File: app/api/custom/role_invites.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        RoleInvite.send_invite_mail(role_invite)
+        return jsonify(
+            status=True,
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:21]: What does `status=True` mean? `success=True`

File: app/api/custom/role_invites.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+            message="Resend invite successfully",
```
Comment (iamareebjamal) [inline app/api/custom/role_invites.py:22]:
```suggestion
            message="Invite resent successfully",
```

File: app/models/role_invite.py [hunk 9]
```diff
@@ -47,3 +52,21 @@ def __repr__(self):
         return '<RoleInvite {!r}:{!r}:{!r}>'.format(
             self.email, self.event_id, self.role_id
         )
+
```
Comment (iamareebjamal) [inline app/models/role_invite.py:55]:
```suggestion
```

File: app/models/role_invite.py (hunk 5 above)
Comment (iamareebjamal) [reply app/models/role_invite.py:57]: Move function above __repr__

File: app/models/role_invite.py [hunk 10]
```diff
@@ -43,6 +49,27 @@ def has_expired(self):
...
+        """
+        Send mail to invitee
+        """
+        user = User.query.filter_by(email=self.email).first()
+        event = Event.query.filter_by(id=self.event_id).first()
+        frontend_url = get_settings()['frontend_url']
+        link = "{}/e/{}/role-invites?token={}".format(
+            frontend_url, event.identifier, self.hash
+        )
+        if has_access('is_coorganizer', event_id=event.id):
```
Comment (iamareebjamal) [inline app/models/role_invite.py:62]:
```python
if not has_access:
    raise ForbiddenError

if user:
    sen...
```

**Review by iamareebjamal — APPROVED** [review]
Great

**Comment by iamareebjamal** [comment]
Please run `pre-commit install` in your virtual environment

**Comment by sachinchauhan2889** [comment]
> Please run `pre-commit install` in your virtual environment

ok sir

**Pull request merged**